cmd/svcat/broker/deregister_cmd.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/command"
+	"github.com/spf13/cobra"
+)
+
+// DeregisterCmd contains the info needed to delete a broker
+type DeregisterCmd struct {
+	BrokerName string
+	Context    *command.Context
+}
+
+// NewDeregisterCmd builds a "svcat deegister" command
+func NewDeregisterCmd(cxt *command.Context) *cobra.Command {
+	deregisterCmd := &DeregisterCmd{
+		Context: cxt,
+	}
+	cmd := &cobra.Command{
+		Use:   "deregister NAME",
+		Short: "Deregisters an existing broker with service catalog",
+		Example: command.NormalizeExamples(`
+		svcat deregister mysqlbroker
+		`),
+		PreRunE: command.PreRunE(deregisterCmd),
+		RunE:    command.RunE(deregisterCmd),
+	}
+	return cmd
+}
+
+// Validate checks that the required arguements have been provided
+func (c *DeregisterCmd) Validate(args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("a broker name is required")
+	}
+	c.BrokerName = args[0]
+
+	return nil
+}
+
+// Run runs the command
+func (c *DeregisterCmd) Run() error {
+	return c.Deregister()
+}
+
+// Deregister calls out to the pkg lib to delete the broker and display the output
+func (c *DeregisterCmd) Deregister() error {
+	err := c.Context.App.Deregister(c.BrokerName)
+	if err != nil {
+		return err
+	}
+
+	outputBytes := []byte(fmt.Sprintf("Successfully removed ClusterServiceBroker \"%v\"", c.BrokerName))
```

Comment:
You can shorten this to `fmt.Fprintf(c.Output, "Successfully removed ClusterServiceBroker %q", c.BrokerName)`

Reply:
nit: I've been avoiding using our type names in the svcat output and pick a user friendly name whenever possible. So I'd prefer it if we didn't go with a type name here. Maybe just go with `Successfully removed broker "foo"` instead, so that when this works for namespaced brokers we can use `Successfully removed broker "foo" from namespace "bar"`.